framework/packages/abstract-testing/src/abstract_mock_querier.rs

```diff
@@ -6,103 +6,68 @@ use abstract_std::{
     },
     version_control::{state::ACCOUNT_ADDRESSES, Account},
 };
-use cosmwasm_std::{testing::MockApi, Addr};
+use cosmwasm_std::Addr;
 use cw_asset::AssetInfo;
 use cw_storage_plus::Item;
 
 use crate::prelude::*;
 
-pub struct AbstractMockQuerierBuilder {
-    builder: MockQuerierBuilder,
-    abstract_addrs: AbstractMockAddrs,
-}
+pub trait AbstractMockQuerier {
+    /// Mock the existence of an Account by setting the Account id for the proxy and manager along with registering the account to version control.
+    fn account(self, account_base: &Account, account_id: AccountId) -> Self;
 
-// ANCHOR: account
-impl AbstractMockQuerierBuilder {
-    pub fn new(mock_api: MockApi) -> AbstractMockQuerierBuilder {
-        AbstractMockQuerierBuilder {
-            builder: MockQuerierBuilder::default(),
-            abstract_addrs: AbstractMockAddrs::new(mock_api),
-        }
-    }
+    /// Add mock assets into ANS
+    fn assets(self, assets: Vec<(&AssetEntry, AssetInfo)>) -> Self;
+
+    fn contracts(self, contracts: Vec<(&ContractEntry, Addr)>) -> Self;
 
+    fn channels(self, channels: Vec<(&ChannelEntry, String)>) -> Self;
+
+    fn addrs(&self) -> AbstractMockAddrs;
```

Comment:
Why do we need this fn if `AbstractMockAddrs` can be constructed easily?

Reply:
We don't need to, but it's useful for other methods of this trait. As we discussed it doesn't have to be minimalistic